### PR TITLE
gh-124787: Fix `TypeAliasType` and incorrect `type_params`

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-124787

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2024-09-30-20-46-32.gh-issue-124787.3FnJnP.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Fix :class:`typing.TypeAliasType` with incorrect ``type_params`` argument.
+Now it raises a :exc:`TypeError` when: 1. A type var without default follows
+a type var with default 2. Not a typelike parameter passed in
+``type_params`` tuple
```
Comment (JelleZijlstra) [inline Misc/NEWS.d/next/Library/2024-09-30-20-46-32.gh-issue-124787.3FnJnP.rst]:
```suggestion
Now it raises a :exc:`TypeError` when a type parameter without a default
follows one with a default, and when an entry in the ``type_params`` tuple
is not a type parameter object.
```

**Review by JelleZijlstra — APPROVED** [review]
Looks good, but let's run this through the refleak buildbots to make sure.

**Comment by JelleZijlstra** [comment]
!buildbot refleak

**Comment by sobolevn** [comment]
Refleaks are green 🎉 
Снимок экрана 2024-10-08 в 13 37 51

**Comment by sobolevn** [comment]
@JelleZijlstra what do you think: is this a bug fix or a new feature? Should we backport this to 3.12 and 3.13?

**Comment by JelleZijlstra** [comment]
To me this feels too risky to backport; let's apply it to 3.14 only.

**Pull request merged**